### PR TITLE
FOGL-1531 - Bad REST payload

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Test result:
`=================================================== 1163 passed, 31 skipped in 85.30 seconds ====================================================
`

### CODE REVIEW

File: python/foglamp/common/microservice_management_client/microservice_management_client.py [hunk 1]
```diff
@@ -232,9 +232,15 @@ def create_configuration_category(self, category_data):
         :param category_data: e.g. '{"key": "TEST", "description": "description", "value": {"info": {"description": "Test", "type": "boolean", "default": "true"}}}'
         :return:
         """
-        url = '/foglamp/service/category'
-
-        self._management_client_conn.request(method='POST', url=url, body=category_data)
+        data = json.loads(category_data)
```
Comment (pintomax) [inline python/foglamp/common/microservice_management_client/microservice_management_client.py]: Is that needed to keep the old behaviour?

File: python/foglamp/services/core/api/configuration.py [hunk 2]
```diff
@@ -99,9 +103,7 @@
         category_desc = data.get('description')
         category_value = data.get('value')
 
-        should_keep_original_items = data.get('keep_original_items', False)
-        if not isinstance(should_keep_original_items, bool):
-            raise TypeError('keep_original_items should be boolean true | false')
+        should_keep_original_items = True if keep_original_items is not None and keep_original_items.lower() in ['true', '1', 1, 't'] else False
```
Comment (pintomax) [inline python/foglamp/services/core/api/configuration.py]: It looks difficult to read.

It can be split in two or more lines

**Comment by amarendra-dianomic** [comment]
@MarkRiddoch rest call to ConfigurationManager.create_category() method is being made solely by microservice management client only and this has been modified in this PR. Rest of all calls to this method are direct calls (e.g. in Scheduler, add_service(), add_task() etc). So, imho, all refereces have been updated. Please correct me if wrong.

File: python/foglamp/services/core/api/configuration.py [hunk 3]
```diff
@@ -84,6 +84,10 @@
     :Example:
             curl -d '{"key": "TEST", "description": "description", "value": {"info": {"description": "Test", "type": "boolean", "default": "true"}}}' -X POST http://localhost:8081/foglamp/category
     """
+    keep_original_items = None
+    if 'keep_original_items' in request.query and urllib.parse.unquote(request.query['keep_original_items']) != '':
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/configuration.py]: @amarendra-dianomic why did we need this `urllib.parse.unquote` for `request.query['keep_original_items']`

File: python/foglamp/services/core/api/configuration.py [hunk 4]
```diff
@@ -84,6 +84,10 @@
...
+        keep_original_items = urllib.parse.unquote(request.query['keep_original_items'])
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/configuration.py]: why unquote?

File: python/foglamp/services/core/api/configuration.py (hunk 2 above)
Comment (praveen-garg) [reply python/foglamp/services/core/api/configuration.py]: @amarendra-dianomic 
https://github.com/foglamp/FogLAMP/pull/955/files#diff-8e6a39d1dfaaf562f1c60b2f201539bfR237
only `keep_original_items.lower() in ['true']`  is needed!

File: python/foglamp/services/core/api/configuration.py (hunk 3 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/configuration.py]: It is a query parameter and, principally, it should always be parsed and validated before using its value. We should not assume that user will always supply the correct and valid value.

File: python/foglamp/services/core/api/configuration.py (hunk 4 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/configuration.py]: See above

File: python/foglamp/services/core/api/configuration.py (hunk 3 above)
Comment (praveen-garg) [reply python/foglamp/services/core/api/configuration.py]: we  allow `true` or `false`  ONLY HENCE NO! We really don't care as those are not valid requests. and should not be respected and check is not required. 
aiohttp router will send us with proper encoding as is to us.

**Pull request merged**